src/commands/env.rs

```diff
@@ -59,50 +73,46 @@ impl Command for Env {
             );
         }
 
-        let shell: Box<dyn Shell> = self
-            .shell
-            .or_else(&infer_shell)
-            .ok_or(Error::CantInferShell)?;
         let multishell_path = make_symlink(config)?;
         let binary_path = if cfg!(windows) {
             multishell_path.clone()
         } else {
             multishell_path.join("bin")
         };
-        println!("{}", shell.path(&binary_path)?);
-        println!(
-            "{}",
-            shell.set_env_var("FNM_MULTISHELL_PATH", multishell_path.to_str().unwrap())
-        );
-        println!(
-            "{}",
-            shell.set_env_var(
-                "FNM_VERSION_FILE_STRATEGY",
-                config.version_file_strategy().as_str()
-            )
-        );
-        println!(
-            "{}",
-            shell.set_env_var("FNM_DIR", config.base_dir_with_default().to_str().unwrap())
-        );
-        println!(
-            "{}",
-            shell.set_env_var("FNM_LOGLEVEL", config.log_level().clone().into())
-        );
-        println!(
-            "{}",
-            shell.set_env_var("FNM_NODE_DIST_MIRROR", config.node_dist_mirror.as_str())
-        );
-        println!(
-            "{}",
-            shell.set_env_var("FNM_ARCH", &config.arch.to_string())
-        );
-        if self.use_on_cd {
-            println!("{}", shell.use_on_cd(config)?);
-        }
-        if let Some(v) = shell.rehash() {
-            println!("{}", v);
+
+        let env_vars = EnvVars{
+            FNM_MULTISHELL_PATH: multishell_path.to_str().unwrap().to_owned(),
+            FNM_VERSION_FILE_STRATEGY: config.version_file_strategy().as_str().to_owned(),
+            FNM_DIR: config.base_dir_with_default().to_str().unwrap().to_owned(),
+            FNM_LOGLEVEL: <&'static str>::from(config.log_level().clone()).to_owned(),
+            FNM_NODE_DIST_MIRROR: config.node_dist_mirror.as_str().to_owned(),
+            FNM_ARCH: config.arch.to_string(),
+        };
+
+        if self.json {
+            println!("{}", serde_json::to_string(&env_vars).unwrap());
+        } else {
+            let shell: Box<dyn Shell> = self
+                .shell
+                .or_else(&infer_shell)
+                .ok_or(Error::CantInferShell)?;
+            println!("{}", shell.path(&binary_path)?);
+
+            println!("{}", shell.set_env_var("FNM_MULTISHELL_PATH", &env_vars.FNM_MULTISHELL_PATH));
+            println!("{}", shell.set_env_var("FNM_VERSION_FILE_STRATEGY", &env_vars.FNM_VERSION_FILE_STRATEGY));
+            println!("{}", shell.set_env_var("FNM_DIR", &env_vars.FNM_DIR));
+            println!("{}", shell.set_env_var("FNM_LOGLEVEL", &env_vars.FNM_LOGLEVEL));
+            println!("{}", shell.set_env_var("FNM_NODE_DIST_MIRROR", &env_vars.FNM_NODE_DIST_MIRROR));
+            println!("{}", shell.set_env_var("FNM_ARCH", &env_vars.FNM_ARCH));
```

Comment:
if `EnvVars` is `HashMap`, we can iterate over the entries and print them one by one, something like

```rust
for (env_var, value) in env_vars.iter() {
    println!("{}", shell.set_env_var(env_var, value));
}
```